app/forms/fields/select_field_with_detail_answer.py

```diff
@@ -28,8 +29,12 @@ def iter_choices(self):
             yield value, label, self.coerce(value) == self.data, detail_answer_id
 
     def pre_validate(self, form):
-        for value, _, _ in self.choices:
-            if value == self.data:
+
+        if not self.choices:
+            raise TypeError(self.gettext("Not a valid choice"))
+
```

Comment:
I probably would want to align with the library now validate_choice is a thing https://github.com/wtforms/wtforms/blob/563b781bb0262ff71dda94f094b5d5413a87af7a/src/wtforms/fields/choices.py#L138

Reply:
the default behaviour is to validate - so you think we should now support the ability to not validate now too? How deeply integrated should this be, considering it's an upgrade?

Reply:
You are right it can be left as is, after all the validate_choice is a new feature we didn't have before and that is the default now. I have a tenancy when overwriting to align as much as possible, but in this instance it can be ignored until we need it.